src/Microsoft.Azure.WebJobs.Host/Listeners/ListenerFactoryListener.cs

```diff
@@ -6,21 +6,24 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Azure.WebJobs.Host.Executors;
+using Microsoft.Azure.WebJobs.Host.Queues;
 
 namespace Microsoft.Azure.WebJobs.Host.Listeners
 {
     internal class ListenerFactoryListener : IListener
     {
         private readonly IListenerFactory _factory;
+        private readonly SharedQueueHandler _sharedQueue;
         private readonly CancellationTokenSource _cancellationSource;
 
         private IListener _listener;
         private CancellationTokenRegistration _cancellationRegistration;
         private bool _disposed;
 
-        public ListenerFactoryListener(IListenerFactory factory)
+        public ListenerFactoryListener(IListenerFactory factory, SharedQueueHandler sharedQueue)
```

Comment:
Shouldn't need a new parameter here. Rather, roll the underlying shared listener into the factory already being passed in.

Reply:
hmm the `IListenerFactory` is an interface that has a lot of implementation, adding field feels difficult
I only wants `sharedQueue` here so that I can control **it's start/stop time against other listeners**:
1. start after all other listeners started (this is not a serious concern, since I don't really know when exactly user of `sharedQueue` will register their handler, the best I can do is to start as late as possible, Technically speaking I can start after other listener's creation has finished)
2. stop after all other listeners have stopped (try to not leave unprocessed in the queue, also a best effort)